src/mpsc/blocking.rs

```diff
@@ -927,6 +1047,127 @@ where
+    ///     thread::sleep(Duration::from_millis(400));
+    ///
+    ///     let mut value = tx.send_ref_timeout(Duration::from_millis(200)).unwrap();
+    ///     write!(value, "world").expect("writing to a `String` should never fail");
+    ///     thread::sleep(Duration::from_millis(400));
+    ///
+    ///     assert_eq!(
+    ///         Err(&SendTimeoutError::Timeout(())),
+    ///         tx.send_ref_timeout(Duration::from_millis(200)).as_deref().map(String::as_str)
+    ///     );
+    /// });
+    /// ```
+    ///
+    /// [`send_timeout`]: Self::send_timeout
+    #[cfg(not(all(test, loom)))]
+    pub fn send_ref_timeout(&self, timeout: Duration) -> Result<SendRef<'_, T>, SendTimeoutError> {
+        send_ref_timeout(
+            &self.inner.core,
+            self.inner.slots.as_ref(),
+            &self.inner.recycle,
+            timeout,
+        )
+    }
+
+    /// Sends a message by value, blocking until there is a free slot to
+    /// write to, for at most `timeout`.
+    ///
+    /// This method takes the message by value, and replaces any previous
+    /// value in the slot. This means that the channel will *not* function
+    /// as an object pool while sending messages with `send_timeout`. This method is
+    /// most appropriate when messages don't own reusable heap allocations,
+    /// or when the [`Receiver`] end of the channel must receive messages by
+    /// moving them out of the channel by value (using the
+    /// [`Receiver::recv`] method). When messages in the channel own
+    /// reusable heap allocations (such as `String`s or `Vec`s), and the
+    /// [`Receiver`] doesn't need to receive them by value, consider using
+    /// [`send_ref_timeout`] instead, to enable allocation reuse.
+    ///
+    ///
+    /// # Errors
+    ///
+    /// - [`Err`]`(`[`SendTimeoutError::Timeout`]`)` if the timeout has elapsed.
+    /// - [`Err`]`(`[`SendTimeoutError::Closed`]`)` if the channel has closed.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use thingbuf::mpsc::{blocking, errors::SendTimeoutError};
+    /// use std::{time::Duration, thread};
+    ///
+    /// let (tx, rx) = blocking::channel(1);
+    ///
+    /// thread::spawn(move || {
+    ///     thread::sleep(Duration::from_millis(500));
+    ///     let msg = rx.recv().unwrap();
+    ///     println!("{}", msg);
+    ///     thread::sleep(Duration::from_millis(500));
+    /// });
+    ///
+    /// thread::spawn(move || {
+    ///     tx.send_timeout(1, Duration::from_millis(200)).unwrap();
+    ///     thread::sleep(Duration::from_millis(400));
+    ///
+    ///     tx.send_timeout(2, Duration::from_millis(200)).unwrap();
+    ///     thread::sleep(Duration::from_millis(400));
+    ///
+    ///     assert_eq!(
+    ///         Err(SendTimeoutError::Timeout(3)),
+    ///         tx.send_timeout(3, Duration::from_millis(200))
+    ///     );
+    /// });
+    /// ```
+    ///
+    /// [`send_ref`]: Self::send_ref
```

Comment:
whoops, this needs to link to `send_ref_timeout`
```suggestion
    /// [`send_ref_timeout`]: Self::send_ref_timeout
```